mathesar/api/serializers/links.py

```diff
@@ -82,6 +82,13 @@ def create(self, validated_data):
     }
     link_type = serializers.CharField(required=True)
 
+    def run_validation(self, data):
```

Comment:
The validation should not be a part of the Mapping Serializer rather it should be for the mapped Serializer.
The validation will fail when creating Many to Many links as the `ManyToManySerializer` does not have a `referent_table` field, you would need to check the `referents` field in that case. So you would need to write a different validation method for each of type of link